common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -539,134 +522,117 @@ public static String bytesToIpAddress(byte[] bytes, int offset, int length) {
         }
     }
 
-    public static boolean isValidIpV6Address(String ipAddress) {
-        int length = ipAddress.length();
-        boolean doubleColon = false;
-        int numberOfColons = 0;
-        int numberOfPeriods = 0;
-        StringBuilder word = new StringBuilder();
-        char c = 0;
-        char prevChar;
-        int startOffset = 0; // offset for [] ip addresses
-        int endOffset = ipAddress.length();
-
-        if (endOffset < 2) {
+    public static boolean isValidIpV6Address(String ip) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "isValidIpV6Address" is 36 which is greater than 25 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)